extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisConfig.java

```diff
@@ -1,77 +1,136 @@
 package io.quarkus.redis.client.runtime;
 
-import java.net.InetSocketAddress;
+import java.net.URI;
 import java.time.Duration;
+import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
 
+import io.quarkus.runtime.annotations.ConfigGroup;
 import io.quarkus.runtime.annotations.ConfigItem;
 import io.quarkus.runtime.annotations.ConfigPhase;
 import io.quarkus.runtime.annotations.ConfigRoot;
 import io.vertx.redis.client.RedisClientType;
+import io.vertx.redis.client.RedisRole;
+import io.vertx.redis.client.RedisSlaves;
 
 @ConfigRoot(phase = ConfigPhase.RUN_TIME)
 public class RedisConfig {
 
     /**
-     * The redis password
+     * The default redis client
      */
-    @ConfigItem
-    public Optional<String> password;
+    @ConfigItem(name = ConfigItem.PARENT)
+    public RedisConfiguration defaultClient;
 
     /**
-     * The redis hosts
+     * Configures additional redis client connections.
+     * <p>
+     * each client have a unique identifier witch must be identified to select the right connection.
+     * example:
+     * <p>
+     *
+     * <pre>
+     * quarkus.redis.client1.hosts = redis://localhost:6379
+     * quarkus.redis.client2.hosts = redis://localhost:6380
+     * </pre>
+     * <p>
+     * And then use annotations above the instances of RedisClient to indicate which instance we are going to use
+     * <p>
```

Comment:
Was it copy/pasted from elsewhere? All this Javadoc block could use some rework.

Reply:
I'll push some updates. Let me know what you think.